subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/execution/DefaultEmptySourceTaskSkipperTest.groovy

```diff
@@ -42,7 +43,7 @@ class DefaultEmptySourceTaskSkipperTest extends Specification {
     final outputChangeListener = Mock(OutputChangeListener)
     final skipper = new DefaultEmptySourceTaskSkipper(cleanupRegistry, TestFiles.deleter(), outputChangeListener, taskInputsListeners)
     final fileCollectionSnapshotter = TestFiles.fileCollectionSnapshotter()
-    final fingerprinter = new AbsolutePathFileCollectionFingerprinter(fileCollectionSnapshotter)
+    final fingerprinter = new AbsolutePathFileCollectionFingerprinter(DirectorySensitivity.FINGERPRINT_DIRECTORIES, fileCollectionSnapshotter)
```

Comment:
I expected `SourceTask` to acquire the new annotation. I think it will have some significant implications to backwards compatibility that in turn can have consequences in how we want to implement this. For that reason it would be good to explore those effects in this PR.

Reply:
I'd personally rather explore that in a follow up PR.  I think this PR provides useful functionality on its own and doesn't drag along the baggage of potentially breaking bits.  I'd rather have that plumbing in place first and then confront the problem of `SourceTask` and what we do about it.  If you feel strongly about it, I'm happy to be convinced otherwise, though.